R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -2165,6 +2165,14 @@ test_that("SQL error message is returned from JVM", {
   expect_equal(grepl("blah", retError), TRUE)
 })
 
+test_that("Default warehouse dir should be set to tempdir", {
+  # nothing should be written outside tempdir() without explicit user permission
+  inital_working_directory_files <- list.files()
```

Comment:
From my test, the `spark-warehouse` directory is created when I run `a <- createDataFrame(iris)`